### ISSUE TITLE
What happens with <img alt="" aria-label="foo" ...>

### ISSUE THREAD

**Issue by DavidMacDonald** [w3c/accname#45]
We got a question on WCAG that may be more appropriate over here. Thoughts?

[w3c/wcag#521](https://github.com/w3c/wcag/issues/521)

I'm guessing the ACCNAME is "foo" and the alt="" is ignored and we should use that algorithm to determine WCAG conformance.

**Comment by accdc** [comment]
Hi,
According to the AccName spec, aria-label always takes precedence directly after aria-labelledby, so if one or the other is supplied, then this will result in the accessible name, no matter what the native host semantic is for that element.

You can test this out by pasting the markup below into the AccName Prototype test page at
https://whatsock.github.io/w3c-alternative-text-computation/Editable%20Live%20Input%20AccName%20Test.html

```
<img alt="" aria-label="Foo" id="test" src="http://whatsock.com/img/whatsock.svg" />
```

Basically, an image with no alt attribute, or one with a null alt attribute (alt=""), is implicitly mapped to role="none", which is why it is not supposed to be exposed to assistive technologies in the accessibility tree.

However, any inclusion of a global attribute such as aria-labelledby or aria-label will override the implied presentational role of that element, and must be exposed to assistive technologies in the accessibility tree regardless.

If this isn't happening within specific browsers by examining the accessibility tree, then bugs need to be filed against those browsers.

Does this help?

All the best,
Bryan

**Comment by aleventhal** [comment]
I agree. It's an authoring error but in general the user agent should be
flexible and handle errors gracefully. The simple rule I remember is that
Aria overrides native. Unless that's changed and I wasn't aware, your
statement is correct.

Aaron

**Comment by accdc** [comment]
Hi David, we discussed this during today's meeting and agreed to close this one for now. If you disagree and want to reopen it as a continuing issue though, please feel free. All the best, Bryan
